lib/vyasa_web/live/media_live/media_bridge.ex

```diff
@@ -259,44 +264,105 @@ defmodule VyasaWeb.MediaLive.MediaBridge do
       sizes: "480x360"
     }
 
-    # generated_artwork = %{
-    #   src:
-    #    "https://i.ytimg.com/vi/AETFvQonfV8/hqdefault.jpg",
-    #   type: "image/jpeg",
-    #   sizes: "480x360"
-    # }
+    playback = Playback.create_playback(loaded_voice, generated_artwork)
 
-    updated_artwork =
-      cond do
-        artwork && is_list(artwork) -> [generated_artwork | artwork]
-        true -> [generated_artwork]
-      end
+    socket
+    |> assign(voice: loaded_voice)
+    |> assign(ack_val: ack_val)
+    |> assign(video: video)
+    |> assign(playback: playback)
+  end
+
+  defp dispatch_voice_registering_events(
+         %Socket{
+           assigns: %{
+             voice:
+               %Voice{
+                 events: voice_events
+               } = _voice,
+             playback: playback
+           }
+         } = socket
+       ) do
+    socket
+    |> push_event("media_bridge:registerEventsTimeline", %{
+      voice_events: voice_events |> create_events_payload()
+    })
+    |> push_event("media_bridge:registerPlayback", %{playback: playback})
+  end
+
+  # TODO: consolidate other hook events that need to be sent to the media bridge hook
+  defp push_hook_events(
+         %Socket{
+           assigns: %{
+             playback:
+               %Playback{
+                 playing?: playing?,
+                 elapsed: elapsed
+               } = playback
+           }
+         } = socket
+       ) do
+    # TODO: merge into a single push_event, let the hook use the Playback::elapsed to determine where to start playing from.
+    socket
+    |> push_event("media_bridge:play_pause", %{
+      cmd:
+        cond do
+          playing? ->
+            "play"
 
-    playback_meta = %Meta{
-      title: title,
-      artists: artists,
-      album: album,
-      artwork: updated_artwork,
-      duration: duration,
-      file_path: file_path
-    }
+          !playing? ->
+            "pause"
+        end,
+      originator: "MediaBridge",
+      playback: playback
+    })
+    |> push_event("media_bridge:seekTime", %{
+      seekToMs: elapsed,
+      originator: "MediaBridge"
+    })
+  end
 
-    IO.inspect(meta, label: "Checkpoint: voice meta:")
-    IO.inspect(playback_meta, label: "Checkpoint: playback meta:")
+  @impl true
+  # On receiving a voice_ack, the written and player contexts are now synced.
+  # The ack_val is used to check if the voice_ack is duplicate, in which case, it's ignored.
+  #
+  # If the ack is fresh, then we shall pipe it to receive_voice_ack() where in
+  # a playback struct is created that represents this synced-state and the client-side hook is triggerred
+  # to register the associated events timeline.
+  def handle_info(
+        {_, :voice_ack,
+         %Voice{
+           video: _video
+         } = voice, ack_val} = _msg,
+        %Socket{
+          assigns: %{
+            ack_val: prev_ack_val
+          }
+        } = socket
+      ) do
+    is_new_voice = ack_val !== prev_ack_val
```

Comment:
ack_val is embedded in the struct and %Voice is already pattern matched 

so a simple curr_voice would work enough which also has a benefit as a more explicit resource lock

mediaplayers only play voice so resources are non-blocking. no need for {:voices, voice.id}. {:video, video.id} etc